include/nlohmann/detail/json_custom_base_class.hpp

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <type_traits>
+
+namespace nlohmann
+{
+namespace detail
+{
+
+struct json_default_base {};
+
+template<class T>
```

Comment:
Please add a document why you need `json_base_class`.

Reply:
see 596e278